client/core/bond.go

```diff
@@ -580,11 +573,38 @@ func (c *Core) rotateBonds(ctx context.Context) {
 			continue
 		}
 
-		recordReserveAdjustments(dc, acctBondState, bondAsset, netReserveDeltas, actuatedTierChanges)
+		if acctBondState.tierChange != 0 {
+			reserveDelta := bondOverlap * -acctBondState.tierChange * int64(bondAsset.Amt)
+			netReserveDeltas[acctBondState.bondAssetID] += reserveDelta // negative means free reserves
+			// After applying the net reserveDelta, we would normally *zero* the
+			// tierChange field, but it could change, so we remember it.
+			actuatedTierChanges[acctBondState.bondAssetID] = append(actuatedTierChanges[acctBondState.bondAssetID],
+				acctUpdate{dc.acct, acctBondState.tierChange, reserveDelta},
+			)
+		}
+
 		c.postRequiredBonds(dc, bondCfg, acctBondState, bondAsset, wallet, expiredStrength, unlocked)
 	}
 
-	c.applyReserveAdjustments(netReserveDeltas, actuatedTierChanges)
+	for assetID, reserveDelta := range netReserveDeltas {
+		if reserveDelta != 0 { // net adjustment across all dexConnections
+			wallet, err := c.connectedWallet(assetID)
+			if err != nil { // we grabbed it above so shouldn't happen
+				c.log.Errorf("%v wallet not available for bond: %v", unbip(assetID), err)
+				continue
+			}
+			c.log.Infof("Updating bond reserves by %s (automatic tier change adjustments)",
+				wallet.amtStringSigned(reserveDelta))
+			wallet.ReserveBondFunds(reserveDelta, false)
+		}
+		// Update the unactuated tier change counter.
+		for _, s := range actuatedTierChanges[assetID] {
```

Comment:
Thanks for switching this back.  The `netReserveDeltas` and `actuatedTierChanges` variables are just naturally locally scoped, and the helper methods shouldn't be used in any other context, so they are baggage and make it hard to see what's really being done.  I think these lines are not so long that it makes `rotateBonds` unreadable for you.  I respect the differences of opinion, and I do think the other refactoring mostly helps.

Reply:
Oh, looks like you need to remove the two helper functions.